tests/python/rest_api/test_tus_uploads.py

```diff
@@ -0,0 +1,254 @@
+# Copyright (C) CVAT.ai Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import base64
+from http import HTTPStatus
+
+import pytest
+import requests
+
+from shared.utils.config import BASE_URL, USER_PASS, make_api_client
+from shared.utils.helpers import generate_image_file
+
+
+@pytest.mark.usefixtures("restore_db_per_function")
+@pytest.mark.usefixtures("restore_cvat_data_per_function")
+@pytest.mark.usefixtures("restore_redis_ondisk_per_function")
+@pytest.mark.usefixtures("restore_redis_inmem_per_function")
+class TestTUSUpload:
+    """Integration tests for TUS resumable upload protocol"""
+
+    _USERNAME = "admin1"
+
+    def _create_task_via_api(self, task_spec):
+        """Helper to create a task"""
+        with make_api_client(self._USERNAME) as api_client:
+            (task, response) = api_client.tasks_api.create(task_spec)
+            assert response.status == HTTPStatus.CREATED
+            return task
+
+    def _start_upload(self, task_id, file_size, filename):
+        """Start TUS upload and return file_id"""
+        metadata = f"filename {base64.b64encode(filename.encode()).decode()}"
+
+        response = requests.post(
+            f"{BASE_URL}/api/tasks/{task_id}/data",
+            headers={
+                "Upload-Length": str(file_size),
+                "Upload-Metadata": metadata,
+                "Tus-Resumable": "1.0.0",
+            },
+            auth=(self._USERNAME, USER_PASS),
+        )
+
+        assert response.status_code == HTTPStatus.CREATED
+        location = response.headers.get("Location")
+        assert location is not None
+        file_id = location.split("/")[-1]
+        return file_id
+
+    def _upload_chunk(self, task_id, file_id, offset, data):
+        """Upload a chunk via TUS PATCH request"""
+        response = requests.patch(
+            f"{BASE_URL}/api/tasks/{task_id}/data/{file_id}",
+            headers={
+                "Upload-Offset": str(offset),
+                "Content-Type": "application/offset+octet-stream",
+                "Tus-Resumable": "1.0.0",
+            },
+            data=data,
+            auth=(self._USERNAME, USER_PASS),
+        )
+        return response
+
+    def _get_upload_offset(self, task_id, file_id):
+        """Get current upload offset via TUS HEAD request"""
+        response = requests.head(
+            f"{BASE_URL}/api/tasks/{task_id}/data/{file_id}",
+            headers={"Tus-Resumable": "1.0.0"},
+            auth=(self._USERNAME, USER_PASS),
+        )
+        assert response.status_code == HTTPStatus.OK
+        offset = response.headers.get("Upload-Offset")
+        assert offset is not None
+        return int(offset)
+
+    def test_can_upload_file_via_tus_in_single_chunk(self):
+        """Test uploading a complete file in one chunk"""
+        task = self._create_task_via_api(
+            {
+                "name": "test TUS single chunk upload",
+                "labels": [{"name": "car"}],
+            }
+        )
```

Comment:
Perhaps you could move this into a fixture? It seems like most of these tests basically create the same task. I would also suggest to check if there's an existing fixture that'll work.